WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -262,6 +262,64 @@ def edit
     render :edit
   end
 
+  def payment_setup
+    @competition = Competition.find(params[:id])
+
+    client = create_oauth_client
+    oauth_params = {
+      scope: 'read_write',
+      state: @competition.id,
+    }
+    @authorise_url = client.auth_code.authorize_url(oauth_params)
+  end
```

Comment:
> And can it really do any damage in the OAuth code?

No idea, but I'd rather not have to think about it!

> Anyway, using Competition.exists?(params[:id]) is probably the right thing here.

That will ultimately be more work though, right? You'll have to explicitly handle the "it does not exist" case, whereas with find, a 404 automatically happens.